buildcmake

```diff
@@ -378,12 +383,65 @@ eval processArgs "$@"
 ############ End option parsing
 
 
+# Make adjustments to $target if necessary
+
 if [[ "$target" = "${target%AMPI-only}AMPI-only" ]]; then
     opt_ampi_only=1
 
     # strip "-only" suffix from AMPI target
     target="${target%AMPI-only}AMPI"
+fi
+
+
+# Determine the build directory name
+
+# Append certain features and compilers to the end of $builddir
+builddir_extra=""
+for flag in opt_omp opt_smp opt_tcp opt_pthreads opt_pxshm opt_syncft opt_ooc opt_persistent opt_cuda; do
+  [[ $flag -eq 1 ]] && builddir_extra+="-${flag/opt_/}"
+done
+
+for c in $opt_lrts_pmi $opt_compiler; do
+  [[ -n $c ]] &&  builddir_extra+="-${c}"
+done
+
+if [[ -n $opt_destination ]]; then
+  # Note that $builddir_extra is intentionally ignored here
+  builddir=$opt_destination$opt_suffix
+else
+  builddir=$triplet$builddir_extra$opt_suffix
+fi
+
 
+# Check whether configure step has already taken place
+
+if [[ -f "$builddir/Makefile" ]]
+then
+  echo "The configure step has already taken place for the specified build."
+  if [[ $FORCE -eq 0 ]]
+  then
+    echo "To attempt a partial rebuild, run 'make' in \"$builddir\"."
+    echo "To remove the existing data and reconfigure, run this script again with --force."
+    exit 1
+  else
+    echo "Removing existing data and starting over. (--force)"
+  fi
+elif [[ -e "$builddir" ]]
```

Comment:
why `-e` and not `-d`? (same below)

Reply:
If someone mistakenly points --destination at an existing file, for example.